pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -508,25 +486,65 @@ func (a *Operator) ensureCSVsInNamespaces(csv *v1alpha1.ClusterServiceVersion, o
 	if err != nil {
 		return err
 	}
+
+	strategyResolver := install.StrategyResolver{}
+	strategy, err := strategyResolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+	if err != nil {
+		return err
+	}
+	strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+	if !ok {
+		return fmt.Errorf("could not cast install strategy as type %T", strategyDetailsDeployment)
+	}
+	ruleChecker := install.NewCSVRuleChecker(a.lister.RbacV1().RoleLister(), a.lister.RbacV1().RoleBindingLister(), a.lister.RbacV1().ClusterRoleLister(), a.lister.RbacV1().ClusterRoleBindingLister(), csv)
+
+	targetCSVs := make(map[string]*v1alpha1.ClusterServiceVersion)
 	for _, ns := range namespaces {
 		if ns.GetName() == operatorGroup.Namespace {
 			continue
 		}
 		if targets.Contains(ns.GetName()) {
-			if err := a.copyToNamespace(csv, ns.GetName()); err != nil {
+			var targetCSV *v1alpha1.ClusterServiceVersion
+			if targetCSV, err = a.copyToNamespace(csv, ns.GetName()); err != nil {
 				a.Log.WithError(err).Debug("error copying to target")
 			}
+			targetCSVs[ns.GetName()] = targetCSV
 		} else {
 			if err := a.pruneFromNamespace(operatorGroup.GetName(), ns.GetName()); err != nil {
 				a.Log.WithError(err).Debug("error pruning from old target")
 			}
 		}
 	}
 
+	targetNamespaces := operatorGroup.Status.Namespaces
+	if targetNamespaces == nil {
+		a.Log.Errorf("operatorgroup '%v' should have non-nil status", operatorGroup.GetName())
+		return nil
+	}
+	for _, ns := range targetNamespaces {
```

Comment:
Does moving this here fix anything in particular? Don't we call:

- ensureCSVsInNamespaces
- ensureTenantRBAC

so this change is basically folding them into one?

Reply:
ensureTenantRBAC has been modified to not use the lister for retrieving the CSV, which is made possible by copyToNamespace returning the CSV.
Also, related to your other point, this keeps RBAC operations for anything other than all namespaces to stay in the syncCopyCSV loop.